src/herbivoresAndCarnivores.js

```diff
@@ -1,15 +1,37 @@
 'use strict';
 
 class Animal {
-  // write your code here
+  static alive = [];
+
+  constructor(name) {
+    this.name = name;
+    this.health = 100;
+    Animal.alive.push(this);
+  }
+
+  static removeDead() {
+    Animal.alive = Animal.alive.filter((animal) => animal.health > 0);
+  }
 }
 
 class Herbivore extends Animal {
-  // write your code here
+  constructor(name) {
+    super(name);
+    this.hidden = false;
+  }
+
+  hide() {
+    this.hidden = true;
+  }
 }
 
 class Carnivore extends Animal {
-  // write your code here
+  bite(target) {
+    if (target instanceof Herbivore && !target.hidden) {
+      target.health -= 50;
+      Animal.removeDead();
+    }
```

Comment:
[CODE PERFOMANCE] - It is preferred to filter `Animal.alive` array inside the `bite` method after the target's health is decreased and becomes less than or equal to 0. Using `Animal.removeDead()` here causes an additional loop through all animals every time a bite occurs, which is not efficient. Consider filtering the `Animal.alive` array directly in this method to remove the dead animals only when necessary.